servicetalk-http-api/src/main/java/io/servicetalk/http/api/FormUrlEncodedHttpSerializer.java

```diff
@@ -168,34 +168,38 @@ private Buffer serialize(@Nullable final Map<String, List<String>> parameters, f
         }
 
         final Buffer buffer = allocator.newBuffer();
-        // Null values may be omitted
-        // https://tools.ietf.org/html/rfc1866#section-8.2
+        // Null values may be omitted, but for more flexibility we choose to encode them if they are provided by the
+        // caller. See https://tools.ietf.org/html/rfc1866#section-8.2
         parameters.forEach((key, values) -> {
             if (key == null || key.isEmpty()) {
                 throw new SerializationException("Null or empty keys are not supported " +
                         "for x-www-form-urlencoded params");
             }
 
-            if (values == null) {
+            if (values == null || values.isEmpty()) {
+                // Received a key with no values, so just encode the key and return.
+                writeKey(buffer, isContinuation, key);
                 return;
             }
 
             values.forEach(value -> {
-                if (value == null) {
-                    return;
+                writeKey(buffer, isContinuation, key);
+                if (value != null) {
+                    buffer.writeBytes("=".getBytes(charset));
```

Comment:
Since you are here, consider caching the value for `UFT-8` and `US_ASCII` encodings. For those we can use `buffer.writeByte(intValueOfTheByte)` and skip allocation of the array.